### PR TITLE
Increase the range for the Calendar start/end year in Fluent. 

### PR DESCRIPTION
**Pull request by lugonzal**
Allow customizing the calendar's reference dates

### Platforms Impacted
- [X] iOS
- [ ] macOS

### Description of changes

Allow customizing the calendar's reference dates, so there's more flexibility for clients to decide what ranges are appropriate for their applications.

### Verification

Make sure date time pickers have a -100 + 100 range. And make sure all usages of these constants are run.

Before
Screen Shot 2020-07-06 at 12 03 45 PM
Screen Shot 2020-07-06 at 12 03 38 PM

After
Screen Shot 2020-07-06 at 12 02 33 PM
Screen Shot 2020-07-06 at 12 02 11 PM

### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/fluentui-apple/pull/109)

### CODE REVIEW

**Comment by Saadnajmi** [comment]
Why isn't this just infinite and lazily loaded?

File: ios/FluentUI/Core/CalendarConfiguration.swift [hunk 1]
```diff
@@ -13,10 +13,8 @@ public typealias MSCalendarConfiguration = CalendarConfiguration
 @objc(MSFCalendarConfiguration)
 open class CalendarConfiguration: NSObject {
     private struct Constants {
-        static let baseReferenceStartTimestamp: TimeInterval = 1420416000 // January 3 2015
-        static let baseReferenceEndTimestamp: TimeInterval = 1736035200 // January 5 2025
```
Comment (Saadnajmi) [inline ios/FluentUI/Core/CalendarConfiguration.swift:17]: Were these constants just unused?

**Comment by lugonzal** [comment]
> Why isn't this just infinite and lazily loaded?

I'm not sure why it was designed this way, I agree this should be infinite, but that'll take some big changes. This change unblocks a current client of the control in the short term.

File: ios/FluentUI/Core/CalendarConfiguration.swift (hunk 1 above)
Comment (lugonzal) [reply ios/FluentUI/Core/CalendarConfiguration.swift:17]: yes

**Comment by lugonzal** [comment]
> > Why isn't this just infinite and lazily loaded?
> 
> I'm not sure why it was designed this way, I agree this should be infinite, but that'll take some big changes. This change unblocks a current client of the control in the short term.

changed the fix a bit, to be more flexible and prevent current clients to have a surprise new range by default.

File: ios/FluentUI/Core/CalendarConfiguration.swift [hunk 2]
```diff
@@ -24,8 +22,11 @@ open class CalendarConfiguration: NSObject {
 
     @objc open var firstWeekday: Int = Calendar.current.firstWeekday
 
-    let referenceStartDate: Date
-    let referenceEndDate: Date
+    /// By default, this is today minus 3 years. If overridden, make sure it's before (less than) the 'referenceEndDate'
+    @objc open var referenceStartDate: Date
+
```
Comment (Saadnajmi) [inline ios/FluentUI/Core/CalendarConfiguration.swift:27]: Could we add a guard in didset, or not because it had @objc ?

Comment (lugonzal) [reply ios/FluentUI/Core/CalendarConfiguration.swift:27]: we could, but that would be assume clients will always set the start date first and then the end date... not sure if that's a correct assumption

**Pull request merged**